From 8c4905abedc2ae7ca507ff2933721ca29ec3a9eb Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@videolan.org>
Date: Wed, 13 Sep 2023 11:19:02 -0400
Subject: [PATCH] Use the pthread library found by CMake in the pkg-config file

It may be empty on some platforms (Android)

See https://cmake.org/cmake/help/latest/module/FindThreads.html for
explanation of the CMake variable.

We already use CMAKE_USE_PTHREADS_INIT to check whether it detected
pthread usage. And HAVE_PTHREAD_H is to to its value.

Change-Id: I93a20ea4f7d0687bfa436de07eac7660c8f1e425
---
 build/cmake/aom_install.cmake | 2 +-
 build/cmake/pkg_config.cmake  | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/build/cmake/aom_install.cmake b/build/cmake/aom_install.cmake
index b02c7b9..2c263e9 100644
--- a/build/cmake/aom_install.cmake
+++ b/build/cmake/aom_install.cmake
@@ -46,12 +46,12 @@ macro(setup_aom_install_targets)
               -DCMAKE_INSTALL_INCLUDEDIR=${CMAKE_INSTALL_INCLUDEDIR}
               -DCMAKE_INSTALL_LIBDIR=${CMAKE_INSTALL_LIBDIR}
               -DCMAKE_PROJECT_NAME=${CMAKE_PROJECT_NAME}
+              -DCMAKE_THREAD_LIBS_INIT=${CMAKE_THREAD_LIBS_INIT}
               -DCONFIG_MULTITHREAD=${CONFIG_MULTITHREAD}
               -DCONFIG_TUNE_VMAF=${CONFIG_TUNE_VMAF}
               -DCONFIG_TUNE_BUTTERAUGLI=${CONFIG_TUNE_BUTTERAUGLI}
               -DCONFIG_SALIENCY_MAP=${CONFIG_SALIENCY_MAP}
               -DCONFIG_TFLITE=${CONFIG_TFLITE}
-              -DHAVE_PTHREAD_H=${HAVE_PTHREAD_H}
               -P
               "${AOM_ROOT}/build/cmake/pkg_config.cmake"
       COMMENT "Writing aom.pc"
diff --git a/build/cmake/pkg_config.cmake b/build/cmake/pkg_config.cmake
index e8fff2e..d3ba67b 100644
--- a/build/cmake/pkg_config.cmake
+++ b/build/cmake/pkg_config.cmake
@@ -13,7 +13,7 @@ cmake_minimum_required(VERSION 3.5)
 set(REQUIRED_ARGS "AOM_ROOT" "AOM_CONFIG_DIR" "CMAKE_INSTALL_PREFIX"
                   "CMAKE_INSTALL_BINDIR" "CMAKE_INSTALL_INCLUDEDIR"
                   "CMAKE_INSTALL_LIBDIR" "CMAKE_PROJECT_NAME"
-                  "CONFIG_MULTITHREAD" "HAVE_PTHREAD_H")
+                  "CONFIG_MULTITHREAD")
 
 foreach(arg ${REQUIRED_ARGS})
   if("${${arg}}" STREQUAL "")
@@ -60,8 +60,8 @@ if(CONFIG_TUNE_BUTTERAUGLI)
 endif()
 file(APPEND "${pkgconfig_file}" "\nConflicts:\n")
 file(APPEND "${pkgconfig_file}" "Libs: -L\${libdir} -l${pkg_name}\n")
-if(CONFIG_MULTITHREAD AND HAVE_PTHREAD_H)
-  file(APPEND "${pkgconfig_file}" "Libs.private: -lm -lpthread\n")
+if(CONFIG_MULTITHREAD AND CMAKE_THREAD_LIBS_INIT)
+  file(APPEND "${pkgconfig_file}" "Libs.private: -lm ${CMAKE_THREAD_LIBS_INIT}\n")
 else()
   file(APPEND "${pkgconfig_file}" "Libs.private: -lm\n")
 endif()
-- 
2.39.2

